server/api/diagnose.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"context"
+	"github.com/juju/errors"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/pd/server"
+	"github.com/unrolled/render"
+	"net/http"
+	"time"
+)
+
+type diagnoseType int
+type diagnoseKV map[diagnoseType]Recommended
+
+//Recommended return detail
```

Comment:
`// Recommended is ...`
btw, `Recommended` is not a good name. Better use noun form `recommendation`, or maybe `Analytic`.